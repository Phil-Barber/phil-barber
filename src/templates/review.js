import React from 'react';
import PageWrapper from '../components/pageWrapper';
import SEO from '../components/seo';
import * as S from './review.styled';

const Details = ({ details }) => (
  <S.DetailsContainer>
    {details.map(({ attr, value }) => (
      <S.Detail key={attr}>
        <strong>{attr}:</strong> {value}
      </S.Detail>
    ))}
  </S.DetailsContainer>
);

export const Review = ({
  imageFluid,
  details,
  title,
  description,
  children,
}) => {
  return (
    <PageWrapper>
      <SEO title={title} description={description} />
      <S.Container>
        <S.InfoColumn>
          <S.Poster fluid={imageFluid} imgStyle={{ objectFit: 'contain' }} />
          <Details details={details} />
        </S.InfoColumn>
        <S.ContentColumn>
          <div>
            <h1>{title}</h1>
            {children}
          </div>
        </S.ContentColumn>
      </S.Container>
    </PageWrapper>
  );
};
