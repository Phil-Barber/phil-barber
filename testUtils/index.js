import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../src/style';
import '@testing-library/jest-dom/extend-expect';

const customRender = (ui, options, store) => {
  const AllTheProviders = ({ children }) => (
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
