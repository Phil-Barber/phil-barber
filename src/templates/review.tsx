import * as React from 'react';
import { useSpring, animated } from 'react-spring';
import PageWrapper from '../components/pageWrapper';
import SEO from '../components/seo';
import * as S from './review.styled';
import { glideIn } from '../style';
import { FluidObject } from 'gatsby-image';

interface Detail {
  readonly attr: string;
  readonly value: string;
}

type Details = ReadonlyArray<Detail>;

interface DetailsProps {
  readonly details: Details;
}

const Details: React.FC<DetailsProps> = ({ details }: DetailsProps) => (
  <S.DetailsContainer>
    {details.map(({ attr, value }) => (
      <S.Detail key={attr}>
        <S.Attr>{attr}:</S.Attr> {value}
      </S.Detail>
    ))}
  </S.DetailsContainer>
);

interface ReviewProps {
  readonly imageFluid: FluidObject;
  readonly details: Details;
  readonly title: string;
  readonly description: string;
  readonly children: React.ReactNode;
  readonly extraDetail?: React.ReactNode;
}

export const Review: React.FC<ReviewProps> = ({
  imageFluid,
  details,
  title,
  description,
  children,
  extraDetail,
}: ReviewProps) => {
  const infoEntrance = useSpring({
    from: { opacity: 0, transform: 'translate3d(-200px, 0, 0)' },
    to: { opacity: 1, transform: 'translate3d(0px, 0, 0)' },
  });
  const contentEntrance = useSpring(glideIn);
  return (
    <PageWrapper>
      <SEO title={title} description={description} />
      <S.Container>
        <S.InfoColumn style={infoEntrance}>
          <S.InfoHeader>
            <S.Poster fluid={imageFluid} imgStyle={{ objectFit: 'contain' }} />
            <Details details={details} />
          </S.InfoHeader>
          <S.ExtraInfo>{extraDetail}</S.ExtraInfo>
        </S.InfoColumn>
        <S.ContentColumn>
          <animated.div style={contentEntrance}>
            <h1>{title}</h1>
            {children}
          </animated.div>
        </S.ContentColumn>
      </S.Container>
    </PageWrapper>
  );
};
