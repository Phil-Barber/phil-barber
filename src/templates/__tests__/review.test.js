import React from 'react';
import { render } from '../../../testUtils';
import { Review } from '../review';

describe('Review', () => {
  it('Matches snapshot', () => {
    const { container } = render(
      <Review
        imageFluid="mockFluid"
        details={[
          { attr: 'Attribute1', value: 'test Value' },
          { attr: 'Attrivbute2', value: 'test value2' },
        ]}
        title="Test title"
        describe="This is a description"
      >
        <div>Child</div>
      </Review>
    );
    expect(container).toMatchSnapshot();
  });
});
