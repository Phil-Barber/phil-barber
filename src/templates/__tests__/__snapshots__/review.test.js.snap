// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Matches snapshot 1`] = `
<div>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
      style="position: relative; top: 0px; left: 0px; right: 0px; z-index: 1; transform: translate3D(0, 0, 0);"
    >
      <div
        class="sc-AxjAm fEzfgO"
      >
        <a
          class="sc-AxgMl ZJAph"
          href="/"
        >
          <h1>
            Test title
          </h1>
        </a>
        <button
          class="sc-AxhUy blLbLW"
        >
          Let it Snow! ðŸ‘ˆ
        </button>
        <span
          class="sc-AxirZ gtVPNy"
          style="transform: rotate(0deg);"
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw jmqPVm"
  >
    <div
      class="sc-Axmtr bdeNGe"
    >
      <div
        class="sc-fzozJi dVHGCq"
      >
        <div
          class="sc-fzoLsD eRrbxT sc-fznZeY flnMgJ"
          style="opacity: 0; transform: translate3d(-200px, 0, 0);"
        >
          <div
            class="sc-fzqBZW kDPVIw"
          >
            <div
              class="sc-fzplWN hNmndR gatsby-image-wrapper"
              style="position: relative; overflow: hidden;"
            >
              <div
                aria-hidden="true"
                style="width: 100%;"
              />
              <picture>
                <source />
                <img
                  alt=""
                  loading="lazy"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: contain; object-position: center; opacity: 0; transition: opacity 500ms;"
                />
              </picture>
              <noscript>
                &lt;picture&gt;&lt;source srcset="undefined" /&gt;&lt;img loading="lazy" src="" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
              </noscript>
            </div>
            <div
              class="sc-fzokOt hbhRWi"
            >
              <span
                class="sc-fznyAO iJRaSg"
              >
                <strong
                  class="sc-fznKkj NwcJf"
                >
                  Attribute1
                  :
                </strong>
                 
                test Value
              </span>
              <span
                class="sc-fznyAO iJRaSg"
              >
                <strong
                  class="sc-fznKkj NwcJf"
                >
                  Attrivbute2
                  :
                </strong>
                 
                test value2
              </span>
            </div>
          </div>
          <div
            class="sc-fzqNJr dgMRDY"
          />
        </div>
        <div
          class="sc-fzoLsD sc-fzpans fMNyIp"
        >
          <div
            style="opacity: 0; transform: translate3d(0, 100px, 0);"
          >
            <h1>
              Test title
            </h1>
            <div>
              Child
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-AxhCb bdrWTY"
    />
  </div>
</div>
`;
